legend-sdlc-server-fs/src/main/java/org/finos/legend/sdlc/server/api/workspace/FileSystemWorkspaceApi.java

```diff
@@ -0,0 +1,339 @@
+                    {
+                        return GROUP_BACKUP_BRANCH_PREFIX;
+                    }
+                    default:
+                    {
+                        throw new RuntimeException("Unknown workspace access type: " + workspaceAccessType);
+                    }
+                }
+            }
+            case USER:
+            {
+                switch (workspaceAccessType)
+                {
+                    case WORKSPACE:
+                    {
+                        return WORKSPACE_BRANCH_PREFIX;
+                    }
+                    case CONFLICT_RESOLUTION:
+                    {
+                        return CONFLICT_RESOLUTION_BRANCH_PREFIX;
+                    }
+                    case BACKUP:
+                    {
+                        return BACKUP_BRANCH_PREFIX;
+                    }
+                    default:
+                    {
+                        throw new RuntimeException("Unknown workspace access type: " + workspaceAccessType);
+                    }
+                }
+            }
+            default:
+            {
+                throw new RuntimeException("Unknown workspace type: " + workspaceType);
+            }
+        }
+    }
+
+    @Override
+    public void deleteWorkspace(String projectId, WorkspaceSpecification workspaceSpecification)
+    {
+        throw new UnsupportedOperationException("Not implemented");
+    }
+
+    @Override
+    public boolean isWorkspaceOutdated(String projectId, WorkspaceSpecification workspaceSpecification)
+    {
+        return false;
+    }
+
+    @Override
+    public boolean isWorkspaceInConflictResolutionMode(String projectId, WorkspaceSpecification workspaceSpecification)
+    {
+        return false;
+    }
+
+    @Override
+    public WorkspaceUpdateReport updateWorkspace(String projectId, WorkspaceSpecification workspaceSpecification)
+    {
+        throw UnavailableFeature.exception();
+    }
+
+    private Collection<? extends Workspace> getBranchesByType(Repository repository, String branchType, String projectId, WorkspaceType wType)
+    {
+        List<Ref> branchesOfType = new ArrayList<>();
+        try
+        {
+            List<Ref> allBranches = Git.wrap(repository).branchList().setListMode(ListBranchCommand.ListMode.ALL).call();
+            for (Ref branch : allBranches)
+            {
+                String type = repository.getConfig().getString("branch", Repository.shortenRefName(branch.getName()), "type");
+                if (type != null && type.equals(branchType))
+                {
+                    branchesOfType.add(branch);
+                }
+            }
+        }
+        catch (GitAPIException e)
+        {
+            LOGGER.error("Error occurred during branch list operation for project {}", projectId, e);
+        }
+        return branchesOfType.stream().map(branch -> workspaceBranchToWorkspace(projectId, branch, wType)).collect(Collectors.toList());
```

Comment:
Why add this map here? Why not just call workspaceBranchToWorkspace when adding to the list initially?